blockprod/src/detail/job_manager/mod.rs

```diff
@@ -72,33 +68,21 @@ pub struct JobHandle {
 )]
 pub struct JobKey {
     /// The job key is provided in tests to make it possible to run
-    /// multiple block productions in parallel
+    /// multiple block productions in parallel.
+    /// In PoS this will be generated based on staker's `stake_public_key` to avoid
+    /// staking twice with the same key.
     custom_id: CustomId,
-    /// The current tip, which will be the "previous block" for the
-    /// block that will be produced
-    current_tip_id: Id<GenBlock>,
-    // TODO: in proof of stake, we also add some identifier of the
-    // current key so that we don't stake twice from the same key.
-    // This is because in PoS, there could be penalties for creating
-    // multiple blocks by the same staker.
 }
 
```

Comment:
FYI: here I've basically moved this `current_tip_id` from the "key" part  of the jobs map to "value", named it `expected_tip_id` and made it optional (the job is not cancelled automatically when the tip changes if `expected_tip_id` is none).

(This is one of those changes that I ended up not making use of, but IMO job_manager looks a bit nicer this way).